### PR TITLE
Map formats

### PR DESCRIPTION
**Pull request by anastasiamac**
Added map-style formatting to pool and volume lists.

### CODE REVIEW

File: cmd/juju/storage/volume.go [hunk 1]
```diff
@@ -35,3 +38,65 @@ func NewVolumeSuperCommand() cmd.Command {
...
+func formatAttachmentInfo(all []params.VolumeAttachment) (map[string]AttachmentInfo, error) {
+	result := make(map[string]AttachmentInfo)
+	for _, one := range all {
+		// TODO (anastasiamac 2015-01-31) add similar logic for volume tags
+		// when available
+		storageTag, err := names.ParseStorageTag(one.Storage)
+		if err != nil {
+			return nil, errors.Annotate(err, "invalid storage tag")
+		}
+		storageName, err := names.StorageName(storageTag.Id())
```
Comment (wallyworld) [inline cmd/juju/storage/volume.go]: could just ignore the error here since it's impossible
storageName, _ := names.StorageName(...)

Comment (anastasiamac) [inline cmd/juju/storage/volume.go]: changed. This approach was copied from the original on "storage list".

File: cmd/juju/storage/volume.go [hunk 2]
```diff
@@ -35,3 +38,62 @@ func NewVolumeSuperCommand() cmd.Command {
...
+	Attachments map[string]AttachmentInfo
+}
+
+type AttachmentInfo struct {
+	Storage     string  `yaml:"storage" json:"storage"`
+	Assigned    bool    `yaml:"assigned" json:"assigned"`
+	Machine     string  `yaml:"machine" json:"machine"`
+	Attached    bool    `yaml:"attached" json:"attached"`
+	DeviceName  string  `yaml:"device-name" json:"device-name"`
+	Size        *uint64 `yaml:"size" json:"size"`
```
Comment (wallyworld) [inline cmd/juju/storage/volume.go]: This should be omitempty if we are making it a pointer and thus allowing it to be nil
The other string fields should also probably be omitempty for completeness, although I think only FileSystem might be ""

File: cmd/juju/storage/poollistformatters.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+			i++
+		}
+		p(name, pool.Type, strings.Join(traits, ","))
+	}
+	tw.Flush()
+
+	return out.Bytes(), nil
+}
+
+type byPoolName []string
```
Comment (axw) [inline cmd/juju/storage/poollistformatters.go]: This is superfluous. Just use `sort.Strings`.

File: cmd/juju/storage/volumelistformatters.go [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+			}
+			p(tag, attachment.Attached, attachment.Machine, attachment.DeviceName, attachmentSize)
+		}
+	}
+	tw.Flush()
+
+	return out.Bytes(), nil
+}
+
+type byVolumeTags []string
```
Comment (axw) [inline cmd/juju/storage/volumelistformatters.go]: ditto

Comment (anastasiamac) [inline cmd/juju/storage/volumelistformatters.go]: changed - thnx :D

File: cmd/juju/storage/poollistformatters.go (hunk 3 above)
Comment (anastasiamac) [inline cmd/juju/storage/poollistformatters.go]: changed - thnx :D

File: cmd/juju/storage/volume.go (hunk 2 above)
Comment (anastasiamac) [inline cmd/juju/storage/volume.go]: Added omitempty to fields and basic tests to exercise when empty.

**Pull request merged**